File: pa2/Lex.c
//-----------------------------------------------------------------------------
//  Jerry Liu
//  jelliu
//  CMPS101
//  pa2
//  Lex.c
//  A test client for the List ADT. Use this to test your list module.
//-----------------------------------------------------------------------------
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "List.h"

#define MAX_LEN 160
int main(int argc, char* argv[])
{
    char line[MAX_LEN];
    FILE *in, *out;

    int lineNumber = 0;

    if( argc != 3 )
    {
        printf("Usage: %s <input file> <output file>\n", argv[0]);
        exit(1);
    }

// open files for reading and writing
    in = fopen(argv[1], "r");
    out = fopen(argv[2], "w");

    if( in==NULL )
    {
        printf("Unable to open file %s for reading\n", argv[1]);
        exit(1);
    }
    if( out==NULL )
    {
        printf("Unable to open file %s for writing\n", argv[2]);
        exit(1);
    }

    while(fgets(line, MAX_LEN, in) != NULL)
    {
        lineNumber++;
    }

    fclose(in);

    char* strArray[lineNumber];
    int i;
    for(i = 0; i < lineNumber; i++)
    {
        strArray[i] = calloc(1, sizeof(char* [MAX_LEN]) );
    }

    in = fopen(argv[1], "r");

    lineNumber = 0;
    while( fgets(line, MAX_LEN, in) != NULL )
    {
        strcpy(strArray[lineNumber], line);
        lineNumber++;
    }
    fclose(in);

    List L = newList();

    for (i = 0; i < lineNumber; i++)
    {
        int isInsert = 0;
        for (moveFront(L); index(L)>= 0; moveNext(L) )
        {
            if (strcmp(strArray[i], strArray[get(L)]) < 0)
            {
                insertBefore(L,i);
                isInsert = 1;
                break;
            }
        }
        if (isInsert == 0)
        {
            append(L,i);
        }
    }

    for (moveFront(L); index(L)>= 0; moveNext(L) )
    {
       fprintf(out, "%s", strArray[get(L)]);
    }
    //printList(out, L);

    return(0);
}

File: pa1/Lex.java
//-----------------------------------------------------------------------------
//  Jerry Liu
//  jelliu ID:1410172
//  CMPS101
//  pa1
//  Lex.java
//  A test client for the List ADT. Use this to test your list module.
//-----------------------------------------------------------------------------

import java.io.*;
import java.util.Scanner;

public class Lex {
	public final static void main(String args[]) throws IOException{
		 Scanner in = null;
	      PrintWriter out = null;
	      
	      int lineNumber = 0;

	      if(args.length < 2) {
	         System.err.println("Usage: FileIO infile outfile");
	         System.exit(1);
	      }
	      
	      in = new Scanner(new File(args[0]));
	      out = new PrintWriter(new FileWriter(args[1]));

	      while( in.hasNextLine() ){
	         lineNumber++;
	         in.nextLine();
	      }
	      
	      in.close();
	      System.out.println("total lines: " + lineNumber);
	      String[] strArray = new String[lineNumber];
	      in = new Scanner(new File(args[0]));
	      lineNumber = 0;
	      while( in.hasNextLine() ){
	    	  strArray[lineNumber] = in.nextLine();
		      lineNumber++;
		  }
	      in.close();

	      
	     
	   
		//String [] strArray = {"one", "two", "three", "four", "five"};
	   
		List l = new List();
		
		for (int i = 0; i < strArray.length; i++) {
			boolean isInsert = false;
			for (l.moveFront(); l.index() >= 0; l.moveNext() ) {
				if (strArray[i].compareTo(strArray[l.get()]) < 0) {
					l.insertBefore(i);
					isInsert = true;
					break;
				}
			}
			if (!isInsert) {
				l.append(i);
			}
		}
		
		System.out.println(l);
		for (l.moveFront(); l.index() >= 0; l.moveNext() ) {
			out.println(strArray[l.get()]);
		}
		out.close();
	}
}

